plugin/CactbotEventSource/CactbotEventSource.cs

```diff
@@ -667,7 +675,9 @@ protected override void Update()
       var result = new JObject();
       result["userLocation"] = config_dir;
       result["localUserFiles"] = user_files == null ? null : JObject.FromObject(user_files);
+
       result["language"] = language_;
+      result["locale"] = pc_locale;
```

Comment:
Sorry for other naming suggestions, but I think "locale" and "language" are a bit too synonymous.

What do you think about:
```
result["parserLanguage"] = language_;
result["systemLanguage"] = pc_locale_;
// For backwards compatibility:
result["language"] = language_;
```

Reply:
I understand some people can be confused with two words, but locale is actually not meaning language.
I sent full locale information because there can be different locale for same language.
example)
`zh-CN` and `zh-TW` use different Chinese character, Simplified one and Traditional one.
I don't know western languages well, so I can't give you another example.

So, to compromise two ideas, I'd better suggest
```
result["parserLanguage"] = language_;
result["systemLocale"] = pc_locale_;
// For backwards compatibility:
result["language"] = language_;
```

Reply:
I was afraid of breaking something by changing variable names, but that backwards compatibility part will make okay with changes.
But I couldn't figure out how javascript codes get variable from c# code, only `resources/user_config.js` is coping with this? And javascript files are so many, and I don't know most of them, so I couldn't change "language" variable name to "parserLanguage" I will break something if I try to do it.
So, those codes are somehow confusing now, if language variable is using Parser's language or System language or even user's setting. I think this should be made clear.

Reply:
My understanding is that locale is a combination of language plus other things.  So, they're not quite the same (and cactbot code interchangeably uses the words locale and language in some places, sorry <_<) but without extra words it's not clear which is the parser and which is the system.  Your suggestion of parserLanguage vs systemLocale sounds great, thanks.

That `result` object automatically becomes `e.detail` here https://github.com/quisquous/cactbot/blob/master/resources/user_config.js#L42.  So if you add more fields, you can just access them by name on `e.detail.systemLocale`.  I think that's the only place that language gets accessed.

Reply:
better understanding of locale is language plus country.
`en-GB` - English, Great Britain
`en-US` - English, United States
`fr-FR` - French, France
`ru-RU` - Russian, Russia
`zh-cn` - zhōngwén(maybe), China

I don't know exact way of making locale string, but it's simply `language-country(or sometimes maybe location)`